tests/lib/SimpleSAML/Utils/XMLTest.php

```diff
@@ -0,0 +1,379 @@
+     * @test
+     */
+    public function testIsDomNodeOfTypeMissingNamespace()
+    {
+        $name = 'name';
+        $namespace_uri = '@missing';
+        $element = new \DOMElement($name, 'value', $namespace_uri);
+
+        XML::isDOMNodeOfType($element, $name, $namespace_uri);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\XML::isDOMNodeOfType
+     * @test
+     */
+    public function testIsDomNodeOfTypeEmpty()
+    {
+        $name = 'name';
+        $namespace_uri = '';
+        $element = new \DOMElement($name);
+
+        $res = XML::isDOMNodeOfType($element, $name, $namespace_uri);
+
+        $this->assertFalse($res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\XML::isDOMNodeOfType
+     * @test
+     */
+    public function testIsDomNodeOfTypeShortcut()
+    {
+        $name = 'name';
+        $namespace_uri = 'urn:oasis:names:tc:SAML:2.0:metadata';
+        $short_namespace_uri = '@md';
+        $element = new \DOMElement($name, 'value', $namespace_uri);
+
+        $res = XML::isDOMNodeOfType($element, $name, $short_namespace_uri);
+
+        $this->assertTrue($res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\XML::isDOMNodeOfType
+     * @test
+     */
+    public function testIsDomNodeOfTypeIncorrectName()
+    {
+        $name = 'name';
+        $bad_name = 'bad name';
+        $namespace_uri = 'ns';
+        $element = new \DOMElement($name, 'value', $namespace_uri);
+
+        $res = XML::isDOMNodeOfType($element, $bad_name, $namespace_uri);
+
+        $this->assertFalse($res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\XML::isDOMNodeOfType
+     * @test
+     */
+    public function testIsDomNodeOfTypeIncorrectNamespace()
+    {
+        $name = 'name';
+        $namespace_uri = 'ns';
+        $bad_namespace_uri = 'bad name';
+        $element = new \DOMElement($name, 'value', $namespace_uri);
+
+        $res = XML::isDOMNodeOfType($element, $name, $bad_namespace_uri);
+
+        $this->assertFalse($res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\XML::getDOMText
+     * @test
+     */
+    public function testGetDomTextBasic()
+    {
+        $data = 'root value';
+        $dom = new \DOMDocument();
+        $element = $dom->appendChild(new \DOMElement('root'));
+        $element->appendChild(new \DOMText($data));
+
+        $res = XML::getDOMText($element);
+        $expected = $data;
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\XML::getDOMText
+     * @test
+     */
+    public function testGetDomTextMulti()
+    {
+        $data1 = 'root value 1';
+        $data2 = 'root value 2';
+        $dom = new \DOMDocument();
+        $element = $dom->appendChild(new \DOMElement('root'));
+        $element->appendChild(new \DOMText($data1));
+        $element->appendChild(new \DOMText($data2));
+
+        $res = XML::getDOMText($element);
+        $expected = $data1 . $data2 . $data1 . $data2;
```

Comment:
Yes, because you feed two times an `DOMText` to the element you're constructing, you've created this document in `$element` after line 134:
```
<?xml version="1.0"?>
<root>root value 1root value 2</root>
```

So the output of `getDOMText` is correct. Whether that's the document you wanted to construct for your test, I don't know.